db.go

```diff
@@ -1507,6 +1534,12 @@ func updateModel(model *Model, exec Executor, list []interface{}) (int64, error)
 		if err != nil {
 			return -1, err
 		}
+		if model.optlockColumnName != nil {
+			if rows != 1 {
+				return -1, errors.New("concurrent modification detected")
```

Comment:
Should this error be factored out into a constant so that callers can handle it more easily?